examples/mediadevices/src/index.js

```diff
@@ -141,9 +143,21 @@ async function applyAudioInputDeviceChange(event) {
 
 // reads selected video input, and updates preview and room to use the device.
 function applyVideoInputDeviceChange(event) {
+  let localVideoTrackPublication = null;
+  if (event) {
+    event.preventDefault();
+    event.stopPropagation();
+  }
+  if(someRoom) {
+    [localVideoTrackPublication] = Array.from(someRoom.localParticipant.videoTracks.values());
+  }
   try {
-    var video = document.querySelector('video#videoinputpreview');
-    applyVideoInputDeviceSelection(deviceSelections.videoinput.value, video, someRoom);
+    const video = document.querySelector('video#videoinputpreview');
+    applyVideoInputDeviceSelection(deviceSelections.videoinput.value, localVideoTrackPublication.track)
```

Comment:
This still doesn't work if you want to change and apply devices before joining a Room. Also, `localVideoTrackPublication.track` will raise an error if you haven't joined the Room (publication will be null).